util/src/validate.ts

```diff
@@ -178,27 +179,44 @@ function checkVertices(ids: string[], nodeModulesPath: string): void {
 		process.stdout.write(`${outputMessage}\r`);
 		count++;
 
-		const validation: ValidatorResult = validateSchema(vertex, vertexSchema);
-		if (!validation.valid) {
-			let errorMessage: string | undefined;
-			vertices[key].invalidate();
+		let errorMessage: string;
+		if (vertex.label === undefined) {
+			errorMessage = `requires property "label"`;
+		} else if (!Object.values(LSIF.VertexLabels).includes(vertex.label)) {
+			errorMessage = `unknown label`;
+		} else {
+			try {
+				let className: string = '';
+				if (vertex.label === LSIF.VertexLabels.event) {
+					if (vertex.scope === undefined) {
+						errorMessage = `requires property "scope"`;
+					} else {
+						className = vertex.scope[0].toUpperCase() + vertex.scope.slice(1) + 'Event';
+					}
+				} else {
+					className = vertex.label[0].toUpperCase() + vertex.label.slice(1);
+				}
 
-			if (vertex.label === undefined) {
-				errorMessage = `requires property "label"`;
-			} else if (!Object.values(LSIF.VertexLabels).includes(vertex.label)) {
-				errorMessage = `unknown label`;
-			} else {
-				try {
-					const className: string = vertex.label[0].toUpperCase() + vertex.label.slice(1);
+				if (schema[className] === undefined) {
 					const specificSchema: TJS.Definition | null = TJS.generateSchema(program, className, { required: true });
-					const moreValidation: ValidatorResult | null = validateSchema(vertex, specificSchema);
-					errorMessage = moreValidation.errors.join('; ');
-				} catch {
-					// Failed to get more details for the error
-					errorMessage = 'unable to provide details';
+					if (specificSchema) {
+						schema[className] = specificSchema;
+					} else {
+						errorMessage = `did not find class ${className}`;
```

Comment:
should you throw here since we wouldn't have any schema?

Reply:
You are absolutely right. This comment made me realize also that I forgot to invalidate vertices with labels undefined or unknown. While it should never be the case that we do not find the class (since we do check if the label is one of the existing ones and only if that's true we proceed to look for the class) it's a good exception to throw in case something goes wrong (say, the protocol changed and now the label does not directly map to a class anymore). Thank you for catching this! I just pushed a fix. 😊